assets/js/types/type-defs/payments.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+export type ObjectType = Record< string, unknown >;
+export type EmptyObjectType = Record< string, never >;
+
+/**
+ * Internal dependencies
+ */
+import type { CartTotals } from './cart';
+import {
+	CartResponseBillingAddress,
+	CartResponseShippingAddress,
+} from './cart-response';
+
+export interface Supports {
+	showSavedCards?: boolean;
+	showSaveOption?: boolean;
+	features?: string[];
+	// Deprecated, in favour of showSavedCards and showSaveOption
+	savePaymentInfo?: boolean;
+}
+
+// we assign a value in the class for supports.features
```

Comment:
I'm wondering if there's a better name than `SupportsInConstructor` or if we even need this type...

Maybe we can:
- Remove this type `SupportsInConstructor`
- Make `Supports.features` mandatory.
- In the `PaymentMethodConfig` `interface` - change the `supports` member to look like this:
```supports: Omit< Supports, 'features' > & { features?: string };```

This is just a suggestion, I would love to hear what you think about this.

Reply:
I don't think `Supports.features` should be made mandatory because this type will also be available for extensions and then there will be a confusion that features needs to be provided in order to register a payment method.

I could have used `	public supports: Supports & { features: string[] };` in the class, but because this type is used in multiple places I still think it's good to keep the `SupportsInConstructor` type in order to avoid repetition. I am opened to renaming it if you want.

Reply:
Hm, ok if `supports.features` is optional then that's fine, but I still think we should change `SupportsInConstructor`'s name. How about renaming like so:

`Supports` -> `SupportsConfiguration`
`SupportsInConstructor` -> `Supports` ?

Reply:
This suggestion for `SupportsConfiguration`  is definitely an improvement. I've also renamed `(Express)PaymentMethodOptions` to `(Express)PaymentMethodConfiguration` for more clarity